inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd

```diff
@@ -43,83 +43,97 @@ ggplot2::theme_set(theme_bw(base_size = 18))
+                              sex = rep.int(c("Male", "Female"), 6))
+population_data_age$population <- as.integer(runif(nrow(population_data_age), 
+                                          min = 500, max = 2000))
 
-# Convert age group variables to numeric variables
-vaccine_data$l1_q10_age_yr <- as.numeric(vaccine_data$l1_q10_age_yr)
-vaccine_data$l1_q55_age_mth <- as.numeric(vaccine_data$l1_q55_age_mth)
+```
 
 
-# Create an age group in months from the year variable
-vaccine_data$age_group <- vaccine_data$l1_q10_age_yr
 
-vaccine_data$age_group[vaccine_data$l1_q55_age_mth < 12 | vaccine_data$l1_q10_age_yr == 0] <- "0-11"
-vaccine_data$age_group[vaccine_data$age_group >= 1 & vaccine_data$age_group < 2] <- "12-23"
-vaccine_data$age_group[vaccine_data$age_group >= 2 & vaccine_data$age_group < 3] <- "24-35"
-vaccine_data$age_group[vaccine_data$age_group >= 3 & vaccine_data$age_group < 4] <- "36-47"
-vaccine_data$age_group[vaccine_data$age_group >= 4 & vaccine_data$age_group < 5] <- "48-59"
-vaccine_data$age_group[vaccine_data$age_group >= 5 & vaccine_data$age_group < 15] <- "60+"
+```{r data cleaning}
+
+# clean up the column names
+colnames(vaccination_raw) <- clean_labels(colnames(vaccination_raw))
 
 
 # Additional variable name cleaning and creation of vaccination status binary variable
-vaccine_data_clean <- vaccine_data %>%
-                      mutate(sex = l1_q5_sex,
-                             vaccination_status = l1_q2_vaccine_9months,
-                             area = l0_q4_settlement,
-                             cluster = l0_q77_what_is_the_cluster_number,
-                             vaccination_status_simple = ifelse(vaccination_status == "Yes - Verbal"
-                                                                | vaccination_status == "Yes - Vaccination card", "Yes",
-                                                                ifelse(vaccination_status == "No" |
-                                                                         vaccination_status == "Don't know", "No", NA)))
+vaccination_clean <- vaccination_raw %>%
+                      mutate(age_in_years = as.integer(q10_age_yr),
+                             age_in_months = as.integer(q55_age_mth),
+                             age_group = sitrep::age_categories(age_in_years, breakers = c(0,1,2,3,4,5)),
+                             sex = q5_sex,
+                             vaccination_routine = q2_vaccine_9months,
+                             vaccination_sia = q32_vaccine_sia,
+                             vaccine_mass = q17_vaccine_mass,
+                             disease_diagnosis = q47_disease_diagnosis,
+                             area = q4_settlement,
+                             cluster = q77_what_is_the_cluster_number)
+                             
+
+```
 
 
+```{r survey weights}
+######## WEIGHTING 
+
+vaccination_clean <- add_weights(vaccination_clean, population_data_age, age_group, sex)
```

Comment:
the issue was that the age_group variable in the population dataset was not a factor - fixed by using the gen_pop function. 
See this [issue post](https://github.com/R4EPI/sitrep/issues/106#issue-426952648)

Reply:
I closed the issue. It was a problem with the specification of the population data. One way to ensure all combinations are represented is to use `expand.grid()`